### DISCUSSION TITLE
additional props to custom plugin to support ssr

### DISCUSSION THREAD

**Discussion by Naveen562117** [react-page/react-page#1050] [Q&A]
in ssr i am loading the data from api and passing as props to my componet which has react page editor which render custom plugins and using global data and i am trying to achive complete preloading of data in custom plugin but the problem is the data which i am getting to my component is not able to reach the custom plugin i have created, example <editor {...props} ..xyz /> so and i have cutom plugin and i would like to access the data in props, is there any way this can be achived,,.. ?? any suggestions would be helpful and thank you in advance

**Comment by macrozone** [comment]
in your custom plugin, you can use any component, hook or whatever in the Renderer Component. Nothing different from your usual react-app. 

I used a lot of apollo-client hooks in my custom cell plugins. You can also use React.createContext / useContext to get data in your cell plugin

**Reply by Naveen562117** [reply]
thank you for the response, the problem is getting the api data on use effect is  not valid option for me because server side rendering, and the props are already loaded and ready to use and I have passed it with the props to editor 

I wanted to figure out if there is any way the additional props attached to editor component are accessible in the plugins

**Reply by macrozone** [reply]
i don't said `useEffect`. Data loading with `useEffect` is usually an anti pattern.

You can use:

- `useContext` & `createContext` to pass data down the tree without passing it as prop to every component. (I advice to learn these advanced react patterns). That is probably what you are looking for.
- apollo-client when dealing with grapqhl-apis (supports ssr as well)
- [swr](https://github.com/vercel/swr/) for rest-apis or anything else.
